report/simple_report.go

```diff
@@ -0,0 +1,250 @@
+	if len(report.RelatedOnionServices) > 0 {
+		out.AddRisk(SEV_MEDIUM, "You are hosting multiple onion services on the same server as this onion service!",
+			"Why this is bad: This may be intentional, but often isn't. Hidden services are best operated in isolation such that a compromise of one does not mean a compromise of the other.",
+			"To fix, host all services on separate infrastructure.",
+			report.RelatedOnionServices)
+	}
+}
+
+// OpenDirectoriesCheck implementation
+type OpenDirectoriesCheck struct{}
+
+func (srt *OpenDirectoriesCheck) Check(out *SimpleReport, report *AnonymityReport) {
+	if len(report.OpenDirectories) > 0 {
+		var severity string
+		var title string
+		if len(report.OpenDirectories) > 10 {
+			severity = SEV_MEDIUM
+			title = "Large number of open directories were discovered!"
+		} else {
+			severity = SEV_LOW
+			title = "Small number of open directories were discovered!"
+		}
+
+		out.AddRisk(severity, title,
+			"Why this is bad: Open directories can reveal the existence of files not linked from the sites source code. Most of the time this is benign, but sometimes operators forget to clean up more sensitive folders.",
+			"To fix, use .htaccess rules or equivalent to make reading directories listings forbidden. Quick Fix (Disable indexing globally) for Debian / Ubuntu running Apache: a2dismod autoindex as root.",
+			report.OpenDirectories)
+	}
+}
+
+// ExifImagesCheck implementation
+type ExifImagesCheck struct{}
+
+func (srt *ExifImagesCheck) Check(out *SimpleReport, report *AnonymityReport) {
+	if len(report.ExifImages) > 0 {
+		var severity string
+		var title string
+		if len(report.OpenDirectories) > 10 {
+			severity = SEV_HIGH
+			title = "Large number of images with EXIF metadata were discovered!"
+		} else {
+			severity = SEV_MEDIUM
+			title = "Small number of images with EXIF metadata were discovered!"
+		}
+		items := []string{}
+		for _, image := range report.ExifImages {
+			items = append(items, image.Location)
+		}
+		out.AddRisk(severity, title,
+			"Why this is bad: EXIF metadata can itself deanonymize a user or service operator (e.g. GPS location, Name etc.). Or, when combined, can be used to link anonymous identities together.",
+			"To fix, re-encode all images to strip EXIF and other metadata.",
+			items)
+	}
+}
+
+// PrivateKeyCheck implementation
+type PrivateKeyCheck struct{}
+
+func (srt *PrivateKeyCheck) Check(out *SimpleReport, report *AnonymityReport) {
+	if report.PrivateKeyDetected {
+		out.AddRisk(SEV_CRITICAL, "Hidden service private key is accessible!",
+			"Why this is bad: This can be used to impersonate the service at any point in the future.",
+			"To fix, generate a new hidden service and make sure the private_key file is not reachable from the web root.",
+			nil)
+	}
+}
+
+// Standard checks performed for SimpleReport generation
+// Plugins can extend this list by calling RegisterSimpleReportCheck
+var checks = []SimpleReportCheck{
+	&EmailAddressCheck{},
```

Comment:
Thinking about putting these in a {name, check} map instead, that would allow passing a list of checks to execute in the input (possibly per host).